### ISSUE TITLE
Need FIPS 140-3 support

### ISSUE THREAD

**Issue by ns-mjagtap** [getsentry/relay#3197]
getsentry/relay v23.12.1

Feature Request: Need FIPS 140-3 support.
CC: @chadwhitacre 

Rationale:
In FedRAMP env, apps must run in FIPS ON mode.
getsentry/relay:23.12.1 uses MD5 algo in ways which are not allowed in FIP ON mode.

Please add FIPS 140-3 support.

**Comment by Dav1dde** [comment]
`md5` is only used for `tag_extraction`: https://github.com/getsentry/relay/blob/602761e74a30e41278d75c932953e923f96740be/relay-event-normalization/src/normalize/span/tag_extraction.rs#L403-L411

This is not a cryptographic use, but we should still consider using something else.

**Comment by jjbayer** [comment]
We cannot simply change the algorithm for computing the span group hash because this would break the user experience (every span group would be split into two groups and the old groups would stick around for 90 days). There might be ways to work around this but it would introduce significant complexity (query time merge of groups, ...).

TL;DR we have to decide whether being compliant with this standard is worth either breaking UX or introducing complexity into our system. cc @chadwhitacre @alexjillard

**Comment by chadwhitacre** [comment]
> This is not a cryptographic use

@ns-mjagtap Is this material for the FedRAMP question?

**Comment by chadwhitacre** [comment]
Cross-linking https://github.com/getsentry/sentry/issues/66195 for Sentry.

**Comment by jjbayer** [comment]
@ns-mjagtap did you actually run into problems with Relay when running it in FIPs mode? Since our use of `md5` is not cryptographic it should not be an issue for compliance.

**Comment by ns-mjagtap** [comment]
I don't have the exact all logs, but the relay container crashed with the following error in FIPS ON mode.

```
ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
```

**Comment by olksdr** [comment]
@ns-mjagtap could you, please, provide the entire relay log from the relay container. All the IDs and secrets can be removed. If you could, please, [enable debug logs](https://docs.sentry.io/product/relay/options/#logging) as well. Thanks

**Comment by olksdr** [comment]
Closing it for now. Please, reopen it when there is more info could be provided.
